packages/core/src/implementations/business/InvitationService.ts

```diff
@@ -135,37 +127,36 @@ export class InvitationService implements IInvitationService {
             );
           });
 
+          // Check to make sure we have the data wallet key
+          if (context.dataWalletKey == null) {
+            return errAsync(
+              new UninitializedError("Data wallet is not initialized!"),
+            );
+          }
+
           // If we are opted in, that wins
-          if (optedInOnChain) {
-            // Check if know about the opt-in in the persistence
+          if (commitmentIndex > -1) {
+            // Check if we know about the opt-in in the persistence
             if (acceptedInvitation != null) {
               // Persistence and chain match!
               return okAsync(EInvitationStatus.Accepted);
             }
+
             // There's no known accepted invitation
-            // Get latest opt-in tokenId from chain, and restore Invitation in the persistence
-            return this.consentRepo
-              .getLatestConsentTokenId(invitation.consentContractAddress)
-              .andThen((tokenIdOrNull) => {
-                if (tokenIdOrNull == null) {
-                  // TODO: This is probably the wrong type of error
-                  return errAsync(
-                    new ConsentContractError("No token ID", null, null),
-                  );
-                }
-                return this.invitationRepo
-                  .addAcceptedInvitations([
-                    new OptInInfo(
-                      invitation.consentContractAddress,
-                      tokenIdOrNull ?? invitation.tokenId,
-                    ),
-                  ])
-                  .map(() => EInvitationStatus.Accepted);
-              });
+            // This is a problem but we'll just add a new accepted invitation
+            return this.dataWalletUtils
+              .deriveOptInInfo(
+                invitation.consentContractAddress,
+                context.dataWalletKey,
+              )
+              .andThen((optInInfo) => {
+                return this.invitationRepo.addAcceptedInvitations([optInInfo]);
```

Comment:
in AcceptedInvitation object (InvitationforStorage) we are not storing any information related to optInInfo so the above call seems not necessary

Reply:
The method is calling this.persistence.updateRecord with the optInInfo, not sure if we really need it or not.